pilot/pkg/serviceregistry/external/conversion.go

```diff
@@ -161,6 +161,25 @@ func convertEndpoint(service *model.Service, servicePort *networking.Port,
 	}
 }
 
+// convertWorkloadInstances translates a WorkloadEntry into ServiceInstances. This logic is largely the
+// same as the ServiceEntry convertInstances.
+func convertWorkloadInstances(wle *networking.WorkloadEntry, services []*model.Service, se *networking.ServiceEntry) []*model.ServiceInstance {
+	out := make([]*model.ServiceInstance, 0)
+	for _, service := range services {
+		for _, port := range se.Ports {
+			ep := convertEndpoint(service, port, wle)
+			// WorkloadEntries always are using mTLS
```

Comment:
@rshriram @smawson is this an assumption we want to make? This is for auto-mtls

Reply:
I don't think so, should be treated just like Pods (some are mTLS and some aren't). In theory we can use WorkloadEntry to add non-proxied endpoints to the mesh.

Reply:
Sven is right. We cant make that assumption. Thats why I added the service account field to the workload entry. If it is present, them we assume mtls. Else, we assume no mtls

Reply:
Some possible options:
* If SA is specified, use mTLS. Otherwise plaintext
* Read the tls label directly, if its missing assume mTLS
* Read the tls label directly, if its missing assume plaintext

Any thoughts? The first seems to make the most sense to me, but on k8s there is the implicit "default" SA which would be missing here

Reply:
I would go with the first one. That is what I had in mind when I created the api as well. Thats how it is also documented

Reply:
ACK, updated

Reply:
I think it is fine to start with the first one and revisit, but we'll have to make sure to document that you always need to fill out the serviceAccount field even if using "default". We also need to document NOT to fill that out if using plaintext. So maybe send a PR to update the comment on that field on WorkloadEntry?

Also longer term we should see if we can automate this, maybe as part of auto-registration?